tools/python/create_nifi_reporting_task.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python3
+"""This is a script to create a ReportingTask in NiFi via the REST API"""
+import sys
+import argparse
+import nipyapi
+
+# no stack trace
+sys.tracebacklimit = 0
+
+
+def init(url: str, username: str, password: str, ca_file: str):
+    """Initialize authenticated connection to NiFi"""
+    nipyapi.config.nifi_config.host = url
+    nipyapi.security.set_service_ssl_context(service='nifi', ca_file=ca_file)
+
+    try:
+        nipyapi.security.service_login(service='nifi', username=username, password=password)
+        print("Successfully authenticated and established connection with [%s]" % url)
+    except Exception as ex:
+        raise Exception("Failed to connect to {}: {}".format(url, str(ex))) from None
+
+
+def find_reporting_task(name: str, port: str):
+    """Find a ReportingTask via its name and port"""
+    flow_api = nipyapi.nifi.apis.flow_api.FlowApi()
+
+    try:
+        reporting_tasks = flow_api.get_reporting_tasks().reporting_tasks
+    except Exception as ex:
+        raise Exception("Could not find ReportingTask[{}/{}]: {}"
+                        .format(name, port, str(ex))) from None
+
+    for task in reporting_tasks:
```

Comment:
We get a list back here, right? Couldn't this be done in one line with filter() or something similar?

Reply:
Yeah im not such a big fan of one liners (pylint isnt either ;)). If i do it in the loop or put a lambda in does not make a big difference to me...